components/BottomNavigation/src/private/MDCBottomNavigationItemView.m

```diff
@@ -0,0 +1,229 @@
+@property(nonatomic, strong) UILabel *label;
+
+@end
+
+@implementation MDCBottomNavigationItemView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonMDCBottomNavigationItemViewInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  self = [super initWithCoder:aDecoder];
+  if (self) {
+    [self commonMDCBottomNavigationItemViewInit];
+  }
+  return self;
+}
+
+- (void)commonMDCBottomNavigationItemViewInit {
+  self.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+
+  _selectedItemTintColor = [UIColor blackColor];
+  _unselectedItemTintColor = [UIColor grayColor];
+
+  _iconImageView = [[UIImageView alloc] initWithFrame:CGRectZero];
+  [self addSubview:_iconImageView];
+
+  _label = [[UILabel alloc] initWithFrame:CGRectZero];
+  _label.text = _title;
+  _label.font = [UIFont systemFontOfSize:kMDCBottomNavigationItemViewTitleFontSize];
+  _label.textAlignment = NSTextAlignmentCenter;
+  _label.textColor = _selectedItemTintColor;
+  [self addSubview:_label];
+
+  _badge = [[MDCBottomNavigationItemBadge alloc] initWithFrame:CGRectZero];
+  [self addSubview:_badge];
+
+  if (!_badgeValue) {
+    _badge.hidden = YES;
+  }
+
+  _circleLayer = [CAShapeLayer layer];
+  CGRect circleLayerRect = CGRectMake(kMDCBottomNavigationItemViewCircleLayerOffset,
+                                      kMDCBottomNavigationItemViewCircleLayerOffset,
+                                      kMDCBottomNavigationItemViewCircleLayerDimension,
+                                      kMDCBottomNavigationItemViewCircleLayerDimension);
+  UIBezierPath *bezierPath = [UIBezierPath bezierPathWithOvalInRect:circleLayerRect];
+  _circleLayer.path = bezierPath.CGPath;
+  _circleLayer.fillColor = _selectedItemTintColor.CGColor;
+  _circleLayer.opacity = 0;
+  [_iconImageView.layer addSublayer:_circleLayer];
+
+  _button = [[UIButton alloc] initWithFrame:self.bounds];
+  _button.autoresizingMask = (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
+  [self addSubview:_button];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+
+  CGFloat labelHeight = [self sizeForText:self.title
+                                     font:self.label.font
+                         boundingRectSize:self.bounds.size].height;
+  self.label.frame = CGRectMake(0, 0, CGRectGetWidth(self.bounds), labelHeight);
+  [self setSelected:self.selected];
+}
+
+- (CGSize)sizeForText:(NSString *)text
+                 font:(UIFont *)font
+     boundingRectSize:(CGSize)boundingRectSize {
+  CGRect rect = [text boundingRectWithSize:boundingRectSize
+                                   options:NSStringDrawingUsesLineFragmentOrigin
+                                attributes:@{ NSFontAttributeName:font }
+                                   context:nil];
+  return rect.size;
+}
+
+- (UIImage *)colorizeImage:(UIImage *)image color:(UIColor *)color {
```

Comment:
nit: This doesn't refer to self, should likely be a class method.
However, we can probably use imageView.tintColor.  (See comment below)

Reply:
In the process of doing research on imageView.tintColor. tintColor does not appear to be affecting icons that are only black with 1.f alpha.

Reply:
Removing this method. Fixed how tintColor is applied to UIImageView.